Code/Framework/AzCore/AzCore/IO/Streamer/Statistics.h

```diff
@@ -10,127 +10,297 @@
+            AZStd::string_view description = "",
+            GraphType graphType = GraphType::Histogram);
+
+        static Statistic CreateTime(
+            AZStd::string_view owner,
+            AZStd::string_view name,
+            TimeValue value,
+            AZStd::string_view description = "",
+            GraphType graphType = GraphType::Lines);
+
+        static Statistic CreateTimeRange(
+            AZStd::string_view owner,
+            AZStd::string_view name,
+            TimeValue value,
+            TimeValue min,
+            TimeValue max,
+            AZStd::string_view description = "",
+            GraphType graphType = GraphType::Lines);
+
+        static Statistic CreateBytesPerSecond(
+            AZStd::string_view owner,
+            AZStd::string_view name,
+            double value,
+            AZStd::string_view description = "",
+            GraphType graphType = GraphType::Histogram);
+
+        static Statistic CreatePermanentString(
+            AZStd::string_view owner, AZStd::string_view name, AZStd::string value, AZStd::string_view description = "");
+        static Statistic CreateReferenceString(
+            AZStd::string_view owner, AZStd::string_view name, AZStd::string_view value, AZStd::string_view description = "");
+        static void PlotImmediate(AZStd::string_view owner, AZStd::string_view name, double value);
+
+        AZStd::string_view GetOwner() const;
+        AZStd::string_view GetName() const;
+        AZStd::string_view GetDescription() const;
+        const Value& GetValue() const;
+        GraphType GetGraphType() const;
 
-            //! Calculates the average value for the window.
-            AverageType CalculateAverage() const
+    private:
+        Statistic() = default;
+
+        template<typename T>
+        static Statistic Create(
+            AZStd::string_view owner, AZStd::string_view name, T&& value, AZStd::string_view description, GraphType graphType);
+
+        Value m_value;
+        AZStd::string_view m_owner;
+        AZStd::string_view m_name;
+        AZStd::string_view m_description;
+        GraphType m_graphType;
+    };
+
+    //! AverageWindow keeps track of the average of values in a sliding window.
+    //! @StorageType The type of the value in the sliding window. Need to be a number of a AZStd::chrono::duration.
+    //! @AverageType The type CalculateAverage will return. StorageType needs to be able to be converted to AverageType.
+    //! @WindowSize The maximum number of entries kept in the window.
+    template<typename StorageType, typename AverageType, size_t WindowSize>
+    class AverageWindow
+    {
+        static_assert(AZStd::is_arithmetic<StorageType>::value || AZStd::chrono::Internal::is_duration<StorageType>::value,
+            "AverageWindow only support numbers and AZStd::chrono::durations."); 
+        static_assert((WindowSize & (WindowSize - 1)) == 0, "The WindowSize of AverageWindow needs to be a power of 2.");
```

Comment:
There was a macro or function that checked power-of-2, wasn't there?

Reply:
There's a `constexpr` function in `MathUtils.h` that will do the trick, so used that.

Reply:
Yes. [IsPowerOfTwo](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Math/MathUtils.h#L70-L73)